enterprise/kernel/src/test/java/org/neo4j/kernel/impl/store/format/highlimit/NodeRecordFormatTest.java

```diff
@@ -123,6 +123,32 @@ public void useVariableLengthFormatWhenPropertyReferenceTooBig() throws IOExcept
         verifySameReferences( source, target);
     }
 
+    @Test
+    public void useVariableLengthFormatWhenRecordSizeIsTooSmall() throws IOException
+    {
+        NodeRecord source = new NodeRecord( 1 );
+        NodeRecord target = new NodeRecord( 1 );
+        source.initialize( true, randomFixedReference(), true, randomFixedReference(), 0L );
+
+        writeReadRecord( source, target, NodeRecordFormat.FIXED_FORMAT_RECORD_SIZE - 1 );
+
+        assertFalse( "Record should use variable length reference if format record is too small.", target.isUseFixedReferences() );
+        verifySameReferences( source, target);
+    }
```

Comment:
If the record is smaller than what can fit in a fixed-width-reference type record, we use a variable-width-reference type record? Why is that? Fixed-width is supposed to be faster for everything that can fit.